### DISCUSSION TITLE
How do I get the public url to S3 uploaded image?

### DISCUSSION THREAD

**Discussion by botzill** [directus/directus#3014] [Archive]
Hi.

Prev version had `full_url` param, the new version seems to not have it. That's not a prob as I can construct that on the fly. 

But the issue is that I created the S3 bucket on DO as `public-read`, but seems that files are uploaded as private? Is there a way to easily get the publicly accessible URL of the uploaded image?

**Comment by rijkvanzanten** [comment]
I would recommend using the `/assets` endpoint for file retrieval, as there's no way to do any sort of limiting / permissions when using the url directly to the upload source. That being said, the filename as stored in your storage provider can be accessed from `directus_files.filename_disk` which you can then append to your full public storage url

**Reply by botzill** [reply]
Well, that is the prob that when I append the file name to my public S3 URLs, it does not work. I think files are uploaded as private and I need to get them via S3 presigned URL. Would be nice to have the possibility to get this presigned URL via API.

**Reply by botzill** [reply]
I see a path like 

```
mydomain.com/assets/1e06f3e7-c027-44ba-91f2-44d7f4c3d0df?key=system-medium-cover&access_token=my_token
```

but that is used for thumbnail, can I use similar to get the original URL?

**Reply by rijkvanzanten** [reply]
No, cause there may or may not be a original URL depending on your storage adapter 🤔

**Reply by botzill** [reply]
Well, the same way we have `/assets` to download that file, we may have to get the URL. For S3 to give the signed URL and for local we can return the path to the local file served. Just an idea. 

For now, I think I will connect to S3 and read the signed URL from there.
